### PR TITLE
Add metric name pluralization guidelines

### PR DESCRIPTION
**Pull request by justinfoote**
This PR is part of the solution to issue #600.

## Changes

This PR is a clarification in the general metrics semantic conventions guidelines. It adds some guidelines about when to use a pluralized metric name (like `system.paging.faults`) and when to use a singular name (like `http.server.duration`).

Related issues #600 and #705

Related oteps: [#108](https://github.com/open-telemetry/oteps/pull/108)

### CODE REVIEW

File: specification/metrics/semantic_conventions/README.md [hunk 1]
```diff
@@ -50,12 +50,29 @@ and confusion for end users. (For example, prefer `runtime.java.gc*` over
...
 OpenTelemetry metadata (e.g. `metric.WithUnit` in Go) SHOULD NOT include the
 units in the metric name. Units may be included when it provides additional
 meaning to the metric name. Metrics MUST, above all, be understandable and
 usable.
 
+### Pluralization
+
+Metric names SHOULD NOT be pluralized, unless the value being recorded
+represents discrete instances of a
+[countable quantity](https://en.wikipedia.org/wiki/Count_noun).
```
Comment (arminru) [inline specification/metrics/semantic_conventions/README.md:65]: Would that statement be correct here?
```suggestion
[countable quantity](https://en.wikipedia.org/wiki/Count_noun),
i.e., its value has a unit of `1`.
```

File: specification/metrics/semantic_conventions/README.md [hunk 2]
```diff
@@ -50,12 +50,29 @@ and confusion for end users. (For example, prefer `runtime.java.gc*` over
...
+Generally, the name SHOULD be pluralized only if the unit of the metric in
+question is a pseudo-unit (like `{faults}` or `{operations}`).
+
+Examples:
```
Comment (arminru) [inline specification/metrics/semantic_conventions/README.md:69]: 👍

**Review by arminru — COMMENTED** [review]
While you're at it, it would be nice if you could add a TOC on top (like on your other PR https://github.com/open-telemetry/opentelemetry-specification/pull/1076/files#r502371390) since the document is already getting quite long. Thanks!

File: specification/metrics/semantic_conventions/README.md (hunk 1 above)
Comment (justinfoote) [reply specification/metrics/semantic_conventions/README.md:65]: _Usually_, but the correlation is not 100%. All of the pluralized names belong to instruments with a unit of `1`, but not all instruments with a unit of `1` have pluralized names.

In system metrics, we have the following non-pluralized names whose units are `1`: `system.process.count`, `system.network.dropped`, `system.*.utilization`.

Comment (arminru) [reply specification/metrics/semantic_conventions/README.md:73]: Ah yes, it was only intended in that direction and not reciprocal. Maybe add
```suggestion
[countable quantity](https://en.wikipedia.org/wiki/Count_noun),
which usually have a unit of `1`.
```
or just leave it out.

File: specification/metrics/semantic_conventions/README.md [hunk 3]
```diff
@@ -1,5 +1,13 @@
 # Metrics Semantic Conventions
 
+<!-- Re-generate TOC with `markdown-toc --no-first-h1 -i` -->
+
+<!-- toc -->
+
+Run the tool to have the TOC generated here
```
Comment (arminru) [inline specification/metrics/semantic_conventions/README.md:7]: Don't forget to run it 😇 
https://github.com/jonschlinkert/markdown-toc

Comment (justinfoote) [reply specification/metrics/semantic_conventions/README.md]: 🤦

File: specification/metrics/semantic_conventions/README.md (hunk 1 above)
Comment (justinfoote) [reply specification/metrics/semantic_conventions/README.md:78]: OK, I'm personally leaning towards just leaving it as it's written. I think this is covered well enough in the next line:
> Generally, the name SHOULD be pluralized only if the unit of the metric in
> question is a pseudo-unit (like `{faults}` or `{operations}`).

But I'm open to adding more if y'all feel strongly about it.

Comment (jmacd) [reply specification/metrics/semantic_conventions/README.md:78]: I :heart: this paragraph. I want to see more about discrete countable quantities and pseudo-units in these specs. 😀

**Review by jmacd — APPROVED** [review]

File: specification/metrics/semantic_conventions/README.md (hunk 1 above)
Comment (justinfoote) [reply specification/metrics/semantic_conventions/README.md:78]: I'm working on the units PR now, and now that it's halfway written, I feel even more strongly that we shouldn't add the bit about units here.  We will have separate guidance that instruments that represent a countable quantities should have a unit of `1` or a pseudo unit.

**Review by MrAlias — APPROVED** [review]

**Comment by justinfoote** [comment]
I've just added one small correction. It turns out, I was mis-using the term pseudo-unit. The things we have in curly braces in the units spec are _non-units_.

**Review by arminru — APPROVED** [review]

**Review by jmacd — APPROVED** [review]

**Comment by jmacd** [comment]
(I'm trying to merge this: @justinfoote surely you've signed the CLA-- not sure what else I can do.)

**Pull request merged**